pkg/config/spec.go

```diff
@@ -565,8 +565,27 @@ func ReadUkiResetSpecFromConfig(c *Config) (*v1.ResetUkiSpec, error) {
 }
 
 func NewUkiInstallSpec(cfg *Config) (*v1.InstallUkiSpec, error) {
+	var activeImg v1.Image
+
+	isoRootExists, _ := fsutils.Exists(cfg.Fs, constants.IsoBaseTree)
+
+	// First try to use the install media source
+	if isoRootExists {
+		activeImg.Source = v1.NewDirSrc(constants.IsoBaseTree)
+	}
+	// Then any user provided source
+	if cfg.Install.Source != "" {
+		activeImg.Source, _ = v1.NewSrcFromURI(cfg.Install.Source)
+	}
+	// If we dont have any just an empty source so the sanitation fails
+	// TODO: Should we directly fail here if we got no source instead of waiting for the Sanitize() to fail?
+	if !isoRootExists && cfg.Install.Source == "" {
+		activeImg.Source = v1.NewEmptySrc()
+	}
```

Comment:
are these assignments stepping on top of each other? i.e. are they in order of relevance (reverted)?

Reply:
yes, this is a copy of the non-uki install spec generation. First the media device as a default, then if there is a explicit source from flags or config we override.

And if there is none, then we generate an empty source and let the sanitize fail due to that.

The empty stuff should probably be fixed, but it needs to be fixed on both installs as it changes the current workflow (let sanitize fail)